internal/pkg/service/buffer/worker/service/conditions.go

```diff
@@ -192,6 +192,22 @@ func (c *checker) watchExports(ctx context.Context, wg *sync.WaitGroup) (errCh <
 			return c.dist.MustCheckIsOwner(event.Value.ReceiverKey.String())
 		}).
 		StartMirroring(wg)
+
+	// Invalidate cache on distribution cache.
+	// See WithFilter above, ownership is changed on distribution change, so cache must be re-generated.
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		for {
+			select {
+			case <-c.dist.OnChangeListener().C:
+				c.exports.Restart()
+			case <-ctx.Done():
+				return
+			}
+		}
+	}()
+
```

Comment:
A tu je 2x bug fix.

V kode hned vyssie je `WithFilter`, ktory obsahuje `c.dist.MustCheckIsOwner`.

Kazdy worker ma na starosti iba cast `receivers`, pre ktore riesi background tasks.
Preto sa do memory cachuju data iba pre tu danu cast.
Vdaka tomu to mozes dobre skalovat, ... ak naskalujes 2x worker na 4x worker, tak spotreba memory na kazdom klesne na `~1/2` (co sa tyka tohto kodu).

Len tu bola jedna zasadna chyba.

Distribucia sa moze, napr. po release zmenit. A chybal tu mechanizmus, ktory by tuto cache invalidovat a znove prehral cely stream + znova vyhodnotil `c.dist.MustCheckIsOwner`.

Reply:
Hodil by sa na to nejaky test. Najprv je potrebne spravit zmeny v pkg, a aby tu toho nebolo vela, tak to davam ako dalsi PR.

Test:
- PR: TODO
- Commit: TODO
- Code: TODO